maps/hazards.go

```diff
@@ -0,0 +1,674 @@
+	}
+}
+
+func (m ExpandingScatterMap) SetupBoard(lastBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
+	return (StandardMap{}).SetupBoard(lastBoardState, settings, editor)
+}
+
+func (m ExpandingScatterMap) UpdateBoard(lastBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
+	err := StandardMap{}.UpdateBoard(lastBoardState, settings, editor)
+	if err != nil {
+		return err
+	}
+
+	currentTurn := lastBoardState.Turn + 1
+	startTurn := 1 // first hazard appears on turn 1
+	spawnEveryNTurns := 2
+
+	// no-op if we're not on a turn that spawns hazards
+	if (currentTurn-startTurn)%spawnEveryNTurns != 0 {
+		return nil
+	}
+
+	// no-op if we have spawned the entire board already
+	if len(lastBoardState.Hazards) == lastBoardState.Width*lastBoardState.Height {
+		return nil
+	}
+
+	rand := settings.GetRand(0)
+
+	startX := rand.Range(1, lastBoardState.Width-1)
+	startY := rand.Range(1, lastBoardState.Width-1)
+
+	if currentTurn == startTurn {
+		editor.AddHazard(rules.Point{X: startX, Y: startY})
+		return nil
+	}
+
+	// determine number of rings in spiral
+	numRings := maxInt(startX, startY, lastBoardState.Width-startX, lastBoardState.Height-startY)
+
+	allPositions := []rules.Point{}
+	for ring := 0; ring < numRings; ring++ {
+		offset := ring + 1
+		positions := []rules.Point{}
+		for x := startX - offset; x < startX+offset+1; x++ {
+			for y := startY - offset; y < startY+offset+1; y++ {
+				if isOnBoard(lastBoardState.Width, lastBoardState.Height, x, y) {
+					if ((x == startX-offset || x == startX+offset) && y >= startY-offset && y <= startY+offset) || ((y == startY-offset || y == startY+offset) && x >= startX-offset && x <= startX+offset) {
+						positions = append(positions, rules.Point{X: x, Y: y})
+					}
+				}
+			}
+		}
+		// shuffle the positions so they are added scattered/randomly
+		rand.Shuffle(len(positions), func(i, j int) {
+			positions[i], positions[j] = positions[j], positions[i]
+		})
+		allPositions = append(allPositions, positions...)
+	}
+
+	chosenPos := currentTurn/spawnEveryNTurns - 1
+	editor.AddHazard(allPositions[chosenPos])
+
+	return nil
+}
+
+type RiverAndBridgesHazardsMap struct{}
+
+func (m RiverAndBridgesHazardsMap) ID() string {
+	return "hz_rivers_bridges"
+}
+
+func (m RiverAndBridgesHazardsMap) Meta() Metadata {
+	return Metadata{
+		Name: "hz_rivers_bridges",
+		Description: `Creates fixed maps that have a lake of hazard in the middle with rivers going in the cardinal directions.
+Each river has one or two 1-square "bridges" over them`,
+		Author: "Battlesnake",
+	}
+}
+
+func (m RiverAndBridgesHazardsMap) SetupBoard(lastBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
+	if err := (StandardMap{}).SetupBoard(lastBoardState, settings, editor); err != nil {
+		return err
+	}
+
+	key := fmt.Sprintf("%dx%d", lastBoardState.Width, lastBoardState.Height)
```

Comment:
optional: Could probably use a `rules.Point` or equivalent as the map key here and avoid using fmt?